src/HttpServer.ts

```diff
@@ -65,31 +65,30 @@ export class HttpServer {
     }
 
     // TODO: use this when changing to jsonrpc
-    async rootHandler(req: any, res: any): Promise<void> {
+    async rootHandler(req: Request, res: Response): Promise<void> {
         let status;
+        let id = -1;
         try {
-            let res;
-            // @ts-ignore
-            const func = this.backend[req.body.method];
-            if (func != null) {
-                res = (await func.apply(this.backend, [req.body.params])) ?? {
-                    code: 200
-                };
-            } else {
-                // @ts-ignore
-                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-                throw Error(
-                    `Implementation of method ${req.body.params} not found on backend!`
-                );
+            if (
+                !Object.prototype.hasOwnProperty.call(req.body, 'id') ||
+                !Object.prototype.hasOwnProperty.call(req.body, 'method')
+            ) {
+                throw Error('Missing properties');
             }
-            status = jsonrpc.success(req.body.id, res);
+            id = req.body.id;
+            const result = (await this.processRootHandler(
+                req.body.method,
+                req.body.params
+            )) ?? { code: 200 };
+            status = jsonrpc.success(id, result);
         } catch (e) {
             if (e instanceof Error) {
                 let stack = e.stack.toString();
                 // remove anything after 'rootHandler'
                 stack = stack.replace(/(rootHandler.*)[\s\S]*/, '$1');
+                stack = stack.replace(/(processRootHandler.*)[\s\S]*/, '$1');
                 status = jsonrpc.error(
-                    req.body.id,
+                    id,
```

Comment:
I don't get this. Why do you want to show id as -1 if the request fails to have a body? (btw also no tests for this)

Reply:
All JsonRpc response needs to include an id, if not it cannot be build. If no id is provided in the request, we need to provide one.